app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt

```diff
@@ -136,28 +134,69 @@ class LyricsBelongGameActivity : AppCompatActivity() {
     /**
      * show the result of lyrics matching
      */
-    private fun checkLyrics(lyricToBeCheck: String, lyrics: String) {
-        findViewById<TextView>(R.id.lyricMatchResult).text = if (lyrics.contains(
-                lyricToBeCheck,
-                ignoreCase = true
-            )
-        ) "res: correct" else "res: too bad"
+    private fun checkLyrics(lyricToBeCheck: String, lyrics: String, gameManager: GameManager) {
```

Comment:
For now, I don't have the answer for your second question. If the speech input from SpeechRecognizer is null, the SpeechRecognizer itself will crash the app, if we call it without initialize, it will crash, and if the speech input is "" (empty) the check pass as correct. 
This is something I need to fix in my next PR.